packages/typespec-ts/test/modularUnit/operations.spec.ts

```diff
@@ -0,0 +1,49 @@
+import { assert } from "chai";
+import { emitModularOperationsFromTypeSpec } from "../util/emitUtil.js";
+import { assertEqualContent } from "../util/testUtil.js";
+
+describe("operations", () => {
+  it("required & optional headers", async () => {
+    const tspContent = `
+      model Bar {
+        prop1: string;
+        prop2: int64;
+      }
+      op read(@header requiredHeader: string, @header optionalHeader?: string, ...Bar): OkResponse;
+        `;
+
+    const operationFiles = await emitModularOperationsFromTypeSpec(tspContent);
+    assert.ok(operationFiles);
+    assert.equal(operationFiles?.length, 1);
+    console.log(operationFiles?.[0]?.getFullText()!);
+    assertEqualContent(
+      operationFiles?.[0]?.getFullText()!,
+      `
+      import { TestingContext as Client } from "../rest/index.js";
+      import { StreamableMethod, operationOptionsToRequestParameters } from "@azure-rest/core-client";
+      
+      export function _readSend(context: Client, requiredHeader: string, body: Bar, options: ReadOptions = { requestOptions: {} }): StreamableMethod<Read200Response> {
+          return context.path("/", ).post({...operationOptionsToRequestParameters(options), 
+          headers: {
+            "required-header": requiredHeader,
+            ...(options?.optionalHeader !== undefined ? {"optional-header": options?.optionalHeader} : {})
```

Comment:
The header definition is a dictionary of `string | number | boolean` items. If the value could be `undefined`, it would cause compile error.

```typescript
headers?: RawHttpHeadersInput;
...
export declare type RawHttpHeadersInput = Record<string, string | number | boolean>;
```
I think the core interface is accurate which only allows valid value as record items. I think its codegen's role to prepare headers correctly.

Reply:
In general, it's recommended to work with strings in HTTP headers because the HTTP specification (RFC 7230) defines header fields as comprising a field name and a string value. While you can represent numbers, booleans, or other data as strings in headers, the native format for header values is string-based.

If we set undefined or null it's a string containing "undefined" or "null" that goes out to the wire. We support numbers and boolean as they can be serialized as strings easily, but on the wire it is strings. 

I would prefer to not support null and if the value is set to undefined not included it. If customers want to pass those values they can do it as strings and would have the same effect if we supported them.

Reply:
Yes, as Jose says HTTP headers are actually always strings. We permit creation of Headers from objects that have numbers and booleans in property fields as a convenience, but we coerce them to strings for the value that goes over the wire: https://github.com/Azure/azure-sdk-for-js/blob/39d6d69284f695287e7a5ce0c42f41da11403ad6/sdk/core/core-rest-pipeline/src/httpHeaders.ts#L40

I agree with Jose's position that the preferred way of handling values that are not one of string/number/boolean is to omit the header from what is sent as part of the request. I think this is a stronger position than what we do today, but has less ambiguity at runtime.

Reply:
I agree to not support type besides string/number/boolean, just have one question:
> I agree with Jose's position that the preferred way of handling values that are not one of string/number/boolean is to omit the header from what is sent as part of the request.

Omit by who ? codegen or core itself ? I feel like we should sperate the interface of accept customer input and the interface we send to the service in the wire layer, which shoud be done by serializer, and we currently don't have one.

Unfortunately, if we have something defined as 
`  op read(@header requiredHeader: string, @header optionalHeader?: string | null, ...Bar): OkResponse;`
typespec won't report any errors. Giving the current strategy is to generate buildable code as long as it is a valid typespec, so either we call out to typespec team or we should figure out a way in the codegen ?

Reply:
I found in openapi3, it has specified that headers could be primitives, arrays, objects here. https://swagger.io/docs/specification/describing-parameters/

Reply:
per https://swagger.io/docs/specification/serialization/ it seems that `Header parameters always use the simple style, that is, comma-separated values.` so arrays/objects are always flattened into a csv of some sort.

Reply:
that's is the serialization, but in the generated code, I don't think we should just generate string at least in Modular ?

Reply:
To clarify something after having some offline discussion with @MaryGao, 
1. For complex arrays and objects, codegen should generate the serialize helper functions, for modular, we should leverage the rlc helper functions to build the object/array as string.
2. For primitive types, boolean/number/string, core can handle this now.
3. We need to generate compilable code if it's undefined or null in typespec, which means we need to have a serializer helper function for undefined or null as there's type mismatch. And because we currently don't have model based overall serialier yet, we will have to put them inline the send request function body.

Reply:
Thanks for Qiaoqiao's summary and I supported the omit logic for both null and undefined in latest pr and feel free to review it again.